aws-android-sdk-iot/src/androidTest/java/com/amazonaws/mobileconnectors/iot/MqttManagerIntegrationTest.java

```diff
@@ -804,37 +806,39 @@ public void onMessageArrived(String topic, byte[] data) {
     }
 
     @Test
-    public void mqttWebSocketReconnect() throws Exception {
-        websocketReconnectionTest(IAM);
+    public void testWebsocketReconnectWithIam() throws Exception {
+        websocketReconnectionTest(AuthenticationMode.IAM);
     }
 
     @Test
-    public void testCustomAuthReconnect() throws Exception {
-        websocketReconnectionTest(CUSTOM_AUTH);
+    public void testWebsocketReconnectWithCustomAuth() throws Exception {
+        websocketReconnectionTest(AuthenticationMode.CUSTOM_AUTH);
     }
 
-    private void websocketReconnectionTest(AWSIotMqttManager.AuthenticationMode authMode) throws InterruptedException {
+    private void websocketReconnectionTest(AuthenticationMode authMode)
+            throws InterruptedException, JSONException {
         final ArrayList<AWSIotMqttClientStatusCallback.AWSIotMqttClientStatus> statuses = new ArrayList<AWSIotMqttClientStatusCallback.AWSIotMqttClientStatus>();
         final ArrayList<String> messages = new ArrayList<String>();
 
-        AWSIotMqttManager mqttManager = new AWSIotMqttManager("int-test-ws-rc", Region.getRegion(Regions.US_EAST_1), endpointPrefix);
+        AWSIotMqttManager mqttManager = new AWSIotMqttManager("int-test-ws-rc",
+                Region.getRegion(Regions.US_EAST_1), endpointPrefix);
 
         mqttManager.setAutoReconnect(true);
-        if (authMode.equals(IAM)) {
+        if (AuthenticationMode.IAM.equals(authMode)) {
             // connect using WebSockets and IAM credentials
             mqttManager.connect(credentialsProvider, new AWSIotMqttClientStatusCallback() {
                 @Override
                 public void onStatusChanged(AWSIotMqttClientStatus status, Throwable throwable) {
                     statuses.add(status);
                 }
             });
-        } else if (authMode.equals(CUSTOM_AUTH)) {
+        } else if (AuthenticationMode.CUSTOM_AUTH.equals(authMode)) {
             // connect using WebSockets and custom authentication token
             mqttManager.connect(
-                    "token-header",
-                    "allow",
-                    "Gv85h71GT07RWo5uWQpqHlYWePN7YVY7KgiuPVEnt65+tDDWyZOlGLIDxvwC7LiBF7FbmG/coeBELKltqmrP0gjM7sIyXlrx315xG3LupoK4bmsNIzg6xxWa0n60WxhOv1p5qXDAhmYWIFuL0enMtuwpCY1W073NqZ/+iOsrJexssZiiLnvVf3z6HReEEER6TfFevrgAEUva0h9pzbUepbJ6RaOO7zZY6e/T/wSHrKl0jA4MalGReKbhdlg0HyTPfILhjorYkF7ew/8Ml+QMr/WU90YMp9Sv8SF0q7WBzbuqxpAR8vxm00rRlgrFdsIRjGFTDmrp9Fsx+xlx0aADCA==",
-                    "CustomAuthorizerIntegTest",
+                    getPackageConfigure().getString("token_key_name"),
```

Comment:
Same here, I'd at least pull out the `getFoo()` call so you just have `config.getString("key_blah")`. For "extra credit" (lol) you could do the "strong modeling" of the keys and options object, too.